extensions/amp-a4a/0.1/real-time-config-manager.js

```diff
@@ -0,0 +1,218 @@
+import {tryParseJson} from '../../../src/json';
+import {dev, user} from '../../../src/log';
+import {Services} from '../../../src/services';
+import {isArray, isObject} from '../../../src/types';
+import {isSecureUrl} from '../../../src/url';
+
+/** @type {string} */
+const TAG = 'real-time-config';
+
+/** @type {number} */
+export const MAX_RTC_CALLOUTS = 5;
+
+/** @enum {string} */
+export const RTC_ERROR_ENUM = {
+  BAD_JSON_RESPONSE: 'bad_json_response',
+  DUPLICATE_URL: 'duplicate_url',
+  INSECURE_URL: 'insecure_url',
+  MAX_CALLOUTS_EXCEEDED: 'max_callouts_exceeded',
+  NETWORK_FAILURE: 'network_failure',
+  UNKNOWN_VENDOR: 'unknown_vendor',
+};
+
+/** @typedef {{
+      rtcResponse: ?Object<string, *>,
+      rtcTime: number,
+      callout: string,
+      error: (RTC_ERROR_ENUM | undefined)}} */
+export let rtcResponseDef;
+
+/**
+ * @param {!Array<Promise<!rtcResponseDef>>} promiseArray
+ * @param {!string} error
+ * @param {!string} callout
+ * @private
+ */
+function logAndAddErrorResponse(promiseArray, error, callout) {
+  dev().warn(TAG, `Dropping RTC Callout to ${callout} due to ${error}`);
+  promiseArray.push(buildErrorResponse(error, callout));
+}
+
+/**
+ * @param {!string} error
+ * @param {!string} callout
+ * @param {number=} opt_rtcTime
+ * @return Promise<{callout: string, error: string, rtcTime: number}>
+ * @private
+ */
+function buildErrorResponse(error, callout, opt_rtcTime) {
+  return Promise.resolve({error, callout, rtcTime: opt_rtcTime || 0});
+}
+
+/**
+ * For a given A4A Element, sends out Real Time Config requests to
+ * any urls or vendors specified by the publisher.
+ * @param {!AMP.BaseElement} a4aElement
+ * @param {!Object<string, !../../../src/service/variable-source.SyncResolverDef>} customMacros The ad-network specified macro
+ *   substitutions available to use.
+ * @return {Promise<!Array<!rtcResponseDef>>|undefined}
+ * @private
+ */
+function maybeExecuteRealTimeConfig(a4aElement, customMacros) {
+  const rtcConfig = validateRtcConfig(a4aElement.element);
+  if (!rtcConfig) {
+    return;
+  }
+  const promiseArray = [];
+  const seenUrls = {};
+  const rtcStartTime = Date.now();
+  // For each publisher defined URL, inflate the url using the macros,
+  // and send the RTC request.
+  (rtcConfig['urls'] || []).forEach(url =>
+    inflateAndSendRtc_(a4aElement, url, seenUrls, promiseArray,
+        rtcStartTime, customMacros,
+        rtcConfig['timeoutMillis'])
+  );
+  // For each vendor the publisher has specified, inflate the vendor
+  // url if it exists, and send the RTC request.
+  Object.keys(rtcConfig['vendors'] || []).forEach(vendor => {
+    const url = RTC_VENDORS[vendor.toLowerCase()];
+    if (!url) {
+      return logAndAddErrorResponse(promiseArray,
+          RTC_ERROR_ENUM.UNKNOWN_VENDOR, vendor);
+    }
+    // The ad network defined macros override vendor defined/pub specifed.
+    const macros = Object.assign(rtcConfig['vendors'][vendor], customMacros);
```

Comment:
I believe this will throw if the rtcConfig for this vendor has not macros.  You can change to Object.assign(rtcConfig['vendors'][vendor] || {}, customMacros)